app/modules/projects/models.py

```diff
@@ -47,8 +61,30 @@ def __repr__(self):
     def members(self):
         return [enrollment.user for enrollment in self.user_membership_enrollments]
 
+    @property
+    def encounters(self):
+        return [member.encounter for member in self.encounter_members]
+
     @db.validates('title')
     def validate_title(self, key, title):  # pylint: disable=unused-argument,no-self-use
         if len(title) < 3:
             raise ValueError('Title has to be at least 3 characters long.')
         return title
+
+    def has_read_permission(self, user, obj):
```

Comment:
Willing to hear arguments otherwise, but I was thinking read permission on project would be super simple.. if you are a member of the project you can read it. This won't get you into any danger because the project itself can't reveal much sensitive data- it's just a bucket. 

I would think anything more fine grained would still be controlled by each encounter and it's owner. 

In the 'old-wildbook' implementation the project view only reveals a link to each encounter, both it's project based and display names if they have been assigned, an option to send it to matching against the project set and a super simple representation of it's latest matching result. 

I expect, since you need to be added to the project by the project owner and each participant can only add encounters they have permission on that this would be sufficient.

Reply:
If it is not expected that there would be some time limit, yeah that parameter should be removed. Such feedback is exactly what I think code inspections should be used for.
The finer grained control in the encounter is what is implemented here. If the obj is not an encounter, this code delegates down to the encounter.
Note I expect this function to be called for asset and submission objects as well as encounters. This function handles the encounters objects and delegates asset and submission handling to the encounter.